parachain/tests/adder.rs

```diff
@@ -61,6 +61,8 @@ impl Externalities for DummyExt {
 }
 
 const TEST_CODE: &[u8] = adder::WASM_BINARY;
+// Code that exposes `validate_block` and loops infinitely
+const INFINITE_LOOP_CODE: &[u8] = halt::WASM_BINARY;
```

Comment:
this should get its own file -- node that this is called `adder` for testing the `adder` dummy parachain.

Reply:
I'm reusing some of the adder tests to check if after halting another parachain can validate normally. So I moved adder tests into submodule, rather than making separate integration tests binaries.